crates/ev-revm/src/factory.rs

```diff
@@ -73,12 +73,39 @@ impl MintPrecompileSettings {
     }
 }
 
+/// Settings for custom contract size limit with activation height.
+#[derive(Debug, Clone, Copy)]
+pub struct ContractSizeLimitSettings {
+    limit: usize,
+    activation_height: u64,
+}
+
+impl ContractSizeLimitSettings {
+    /// Creates a new settings object.
+    pub const fn new(limit: usize, activation_height: u64) -> Self {
+        Self {
+            limit,
+            activation_height,
+        }
+    }
+
+    const fn activation_height(&self) -> u64 {
+        self.activation_height
+    }
+
+    const fn limit(&self) -> usize {
+        self.limit
+    }
+}
+
 /// Wrapper around an existing `EvmFactory` that produces [`EvEvm`] instances.
 #[derive(Debug, Clone)]
 pub struct EvEvmFactory<F> {
     inner: F,
     redirect: Option<BaseFeeRedirectSettings>,
     mint_precompile: Option<MintPrecompileSettings>,
+    /// Custom contract size limit with activation height.
```

Comment:
nit, I would suggest doc'ing all fields or none here. I think a single comment on the factory struct mentioning the struct contains `Option` fields for optional features would do the job